sympy/solvers/ode.py

```diff
@@ -277,10 +277,10 @@
 #: ``best``, and ``all_Integral`` meta-hints should not be included in this
 #: list, but ``_best`` and ``_Integral`` hints should be included.
 allhints = (
-    "separable",
-    "1st_exact",
     "1st_linear",
     "Bernoulli",
+    "separable",
+    "1st_exact",
```

Comment:
I guess for these, it's pretty obvious that it's better. As I pointed out in the issue, 1st_linear and Bernoulli always return a solution that is solved for f(x), whereas these rely on `solve()` to be able to do the same.